code/modules/admin/verbs/adminhelp.dm

```diff
@@ -535,38 +565,100 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 	deltimer(adminhelptimerid)
 	adminhelptimerid = 0
 
-// Used for methods where input via arg doesn't work
-/client/proc/get_adminhelp()
-	var/msg = input(src, "Please describe your problem concisely and an admin will help as soon as they're able.", "Adminhelp contents") as message|null
-	adminhelp(msg)
+GLOBAL_DATUM_INIT(admin_help_ui_handler, /datum/admin_help_ui_handler, new)
 
-/client/verb/adminhelp(msg as message)
-	set category = "Admin"
-	set name = "Adminhelp"
+/datum/admin_help_ui_handler
+	var/list/ahelp_cooldowns = list()
+
+/datum/admin_help_ui_handler/ui_state(mob/user)
+	return GLOB.always_state
+
+/datum/admin_help_ui_handler/ui_data(mob/user)
+	. = list()
+	var/list/admins = get_admin_counts(R_BAN)
+	.["adminCount"] = length(admins["present"])
+
+/datum/admin_help_ui_handler/ui_static_data(mob/user)
+	. = list()
+	.["bannedFromUrgentAhelp"] = is_banned_from(user.ckey, "Urgent Adminhelp")
+	.["urgentAhelpPromptMessage"] = CONFIG_GET(string/urgent_ahelp_user_prompt)
+	var/webhook_url = CONFIG_GET(string/adminhelp_webhook_url)
+	if(webhook_url)
+		.["urgentAhelpEnabled"] = TRUE
+
+/datum/admin_help_ui_handler/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Adminhelp")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+/datum/admin_help_ui_handler/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+	var/client/user = usr.client
+	var/message = sanitize_text(trim(params["message"]), null)
```

Comment:
```suggestion
	var/message = sanitize_text(trim(params["message"]))
```

`null` won't actually be used in place of the default, it's going to be `""` no matter what.